Mathlib/Geometry/Euclidean/Angle/Unoriented/TriangleInequality.lean

```diff
@@ -0,0 +1,220 @@
+/-
+Copyright (c) 2025 Ilmārs Cīrulis. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Ilmārs Cīrulis
+-/
+import Mathlib.Geometry.Euclidean.Angle.Unoriented.Affine
+
+/-!
+# The Triangle Inequality for Angles
+
+This file contains proof that angles obey the triangle inequality.
+-/
+
+open InnerProductGeometry
+
+open scoped RealInnerProductSpace
+
+variable {V : Type*}
+variable [NormedAddCommGroup V]
+variable [InnerProductSpace ℝ V]
+
+/-- **Gets unit length vector from given vector** -/
+noncomputable def unit (x : V): V := ‖x‖⁻¹ • x
+
+lemma norm_of_unit {x : V} (H : x ≠ 0) : ‖unit x‖ = 1 := by
+  unfold unit; rw [norm_smul];
+  simp only [norm_inv, norm_norm]
+  have H0 : ‖x‖ ≠ 0 := by simp only [ne_eq, norm_eq_zero]; exact H
+  field_simp
+
+lemma inner_product_of_units_as_cos {x y : V} (Hx : ‖x‖ = 1) (Hy : ‖y‖ = 1) :
+    ⟪x, y⟫ = Real.cos (angle x y) := by
+  rw [cos_angle, Hx, Hy]; simp
+
+lemma inner_sq_of_unit {x : V} (Hx : ‖x‖ = 1) : ⟪x, x⟫ = 1 := by
+  rw [inner_eq_one_iff_of_norm_one Hx Hx]
+
+lemma angle_triangle_aux1 {x y : V} (Hx : ‖x‖ = 1) (Hy : ‖y‖ = 1) :
+    let u := unit (x - ⟪x, y⟫ • y)
+    ⟪x, u⟫ ^ 2 + ⟪x, y⟫ ^ 2 = 1 := by
```

Comment:
Using a `let` in the type of a lemma is somewhat unusual. Certainly in a case like this where `u` is only used once in the statement, it should just be expanded inline in the statement rather than using `let` at all.